### ISSUE TITLE
Manual map 2021 false negative

### ISSUE THREAD

**Issue by ghost** [rust-lang/rust-clippy#7876]
Lint name: manual_map

The 2021 edition made closure capturing smarter. We can now make some suggestions that previously wouldn't compile. 
[Edition Docs](https://doc.rust-lang.org/edition-guide/rust-2021/disjoint-capture-in-closures.html)

I tried this code:

This is the test case for issue #6797

```rust
    let x1 = (Some(String::new()), 0);
    let x2 = x1.0;

    match x2 {
        Some(x) => Some((x, x1.1)),
        None => None,
    };
```

I expected to see this happen: 
This recommendation if I'm using 2021.
```rust
    x2.map(|x| Some((x, x1.1)));
```
Instead, this happened:
No, warning. It's a false negative.

---

[playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=4fac664be6e8c63e937d26239ced26af)
